harness/determined/core/_train.py

```diff
@@ -68,6 +68,26 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
+    def set_metadata(self, metadata: Dict[str, Any]) -> None:
+        """
+        Set the metadata on the current run to the Determined master.
+
+        The metadata is a dictionary of key-value pairs that can be used for analysis,
+        post-processing, or debugging.
+        """
+
+        body = {"metadata": metadata}
+        logger.debug(f"log_metadata({metadata})")
+        self._session.post(f"/api/v1/runs/{self._trial_id}/metadata", json=body)
+
+    def get_metadata(self) -> Dict[str, Any]:
+        """
+        Get the metadata of the current run from the Determined master.
+        """
+        r = self._session.get(f"/api/v1/runs/{self._trial_id}/metadata")
+        assert r.status_code == 200, f"failed to get metadata: {r.text}"
+        return dict(r.json())
```

Comment:
response.json() already returns a dict

Reply:
I believe it's returning an Any data type, I was getting errors when using it without the cast initially.

```
(method) def json(
    *,
    cls: type[JSONDecoder] | None = ...,
    object_hook: ((dict[Any, Any]) -> Any) | None = ...,
    parse_float: ((str) -> Any) | None = ...,
    parse_int: ((str) -> Any) | None = ...,
    parse_constant: ((str) -> Any) | None = ...,
    object_pairs_hook: ((list[tuple[Any, Any]]) -> Any) | None = ...,
    **kwds: Any
) -> Any
```

Reply:
oh, why aren't you just calling the bindings method?

Reply:
That's a good point!